### PR TITLE
Deprecate ``compressed`` in DQM

### PR DESCRIPTION
**Pull request by JoelPasvolsky**
Part of https://github.com/dwavesystems/dwave-system/pull/356

### CODE REVIEW

**Comment by JoelPasvolsky** [comment]
Failed `test-osx-py35` looks unrelated to this PR, it is using `dqm.to_file()` but without parameters:
`buff = f.read() != buff2 = bytearray(len(buff))`  for `dqm.to_file()`

**Review by arcondello — APPROVED** [review]

File: dimod/discrete/discrete_quadratic_model.py [hunk 1]
```diff
@@ -663,14 +663,17 @@ def _to_file_numpy(self, file, compressed):
         file.write(np.dtype('<u4').type(end - start).tobytes())
         file.seek(end)
 
-    def to_file(self, compressed=False, ignore_labels=False,
+    def to_file(self, compress=False, compressed=None, ignore_labels=False,
```
Comment (randomir) [inline dimod/discrete/discrete_quadratic_model.py:666]: To preserve function signature for positional args, it's better to just append the new argument. In theory, this change breaks the public API, and would require a major version increment :grimacing:

**Review by randomir — COMMENTED** [review]
LGTM

File: dimod/discrete/discrete_quadratic_model.py (hunk 1 above)
Comment (JoelPasvolsky) [reply dimod/discrete/discrete_quadratic_model.py:666]: Sorry, I figured it's unlikely anyone was already using this, so I preferred the least confusing signature and to ensure a deprecation notice if there is someone using it already.

Comment (randomir) [reply dimod/discrete/discrete_quadratic_model.py:666]: No, absolutely, this was just an FYI comment. I'm pretty sure this change won't break anyone's code.

**Pull request merged**